README.md

```diff
@@ -18,6 +18,124 @@ to facilitate unit-testing of OpenTracing Java instrumentation.
 
 Packages are deployed to Maven Central under the `io.opentracing` group.
 
+## Usage
+
+### Initialization
+
+Initialization is OpenTracing-implementation-specific. Generally speaking, the pattern is to initialize a `Tracer` once
+for the entire process and to use that `Tracer` for the remainder of the process lifetime. The
+[GlobalTracer](https://github.com/opentracing-contrib/java-globaltracer) repository provides a helper for singleton
+access to the `Tracer` as well as `ServiceLoader` support for OpenTracing Java implementations.
+
+### "Active" `Span`s, `Handle`s, and within-process prapagation
+
+For any execution context or Thread, at most one `Span` may be "active". Of course there may be many other `Spans` involved with the execution context which are (a) started, (b) not finished, and yet (c) not "active": perhaps they are waiting for I/O, blocked on a child Span, or otherwise off the critical path.
+ 
+It's inconvenient to pass an active `Span` from function to function manually, so OpenTracing provides an `ActiveSpanSource` abstraction to provide access to the active `Span` and to pin and defer it for re-activation in other execution contexts (e.g., in an async callback).
+
+Every `Tracer` implementation _must_ provide access to a `ActiveSpanSource` (typically provided at `Tracer` initialization time). The `ActiveSpanSource` in turn exposes the active `Span` via an `ActiveSpanSource.Handle`, like so:
+
+```
+io.opentracing.Tracer tracer = ...;
+...
+SpanHandle activeHandle = tracer.spanSource().active();
+if (activeHandle != null) {
+    activeHandle.span().log("...");
+}
+```
+
+### Starting a new Span
+
+The absolutel simplest case – which does not take advantage of `ActiveSpanSource` – looks like this:
+
+```
+io.opentracing.Tracer tracer = ...;
+...
+try (Span span = tracer.buildSpan("someWork").start()) {
```

Comment:
@bhs I suspect that this "simplest case" should now be:
`try (ActiveSpan span = tracer.buildSpan("someWork").startAndActivate()) {`

Below it should also be `ActiveSpan` instead of `ActiveSpanHoldar.Handle`